plenum/server/domain_req_handler.py

```diff
@@ -175,48 +175,9 @@ def get_value_from_state(self, path, head_hash=None, with_proof=False):
         :param get_value: whether to return the value
         :return: a state proof or None
         '''
-        root_hash = head_hash if head_hash else self.state.committedHeadHash
-        encoded_root_hash = state_roots_serializer.serialize(bytes(root_hash))
-
-        if not with_proof:
-            return self.state.get_for_root_hash(root_hash, path), None
-
-        multi_sig = self.bls_store.get(encoded_root_hash)
         if not multi_sig:
-            # Just return the value and not proof
-            try:
-                return self.state.get_for_root_hash(root_hash, path), None
-            except KeyError:
-                return None, None
-        else:
-            try:
-                proof, value = self.state.generate_state_proof(key=path,
-                                                               root=self.state.get_head_by_hash(root_hash),
-                                                               serialize=True,
-                                                               get_value=True)
-                value = self.state.get_decoded(value) if value else value
-                encoded_proof = proof_nodes_serializer.serialize(proof)
-                proof = {
-                    ROOT_HASH: encoded_root_hash,
-                    MULTI_SIGNATURE: multi_sig.as_dict(),
-                    PROOF_NODES: encoded_proof
-                }
-                return value, proof
-            except KeyError:
-                return None, None
+            root_hash = head_hash if head_hash else self.state.committedHeadHash
+            encoded_root_hash = state_roots_serializer.serialize(bytes(root_hash))
 
-    @staticmethod
-    def make_result(request, data, last_seq_no, update_time, proof):
-        result = {**request.operation, **{
-            DATA: data,
-            f.IDENTIFIER.nm: request.identifier,
-            f.REQ_ID.nm: request.reqId,
-            f.SEQ_NO.nm: last_seq_no,
-            TXN_TIME: update_time
-        }}
-        if proof and request.protocolVersion and \
-                request.protocolVersion >= PlenumProtocolVersion.STATE_PROOF_SUPPORT.value:
-            result[STATE_PROOF] = proof
-
-        # Do not inline please, it makes debugging easier
-        return result
+            multi_sig = self.bls_store.get(encoded_root_hash)
```

Comment:
Why can't we move this calculation to base method? The only source of multi-sig is `bls_store` , so it looks like quite generic logic.

Reply:
PoolReqHandler doesn't  work with a bls_store, therefore, the bls_store and work with it were not moved to the LedgerRequestHandler. But I think this is not so important, we can move it.

Reply:
I think it would be great to move it there, but this is fine to do it latter.